src/components/Navbar/SearchDrawer/SearchDrawer.tsx

```diff
@@ -53,46 +57,37 @@ const SearchDrawer: React.FC = () => {
     }
   }, [isOpen, focusInput]);
 
-  // This useEffect is triggered when the debouncedSearchQuery value changes
-  // eslint-disable-next-line react-func/max-lines-per-function
+  /**
+   * Call BE to fetch the results using the passed filters.
+   *
+   * @param {string} query
+   * @param {number} page
+   */
+  const getResults = useCallback(
+    (query: string, page: number) => {
+      searchGetResults(
+        SearchQuerySource.SearchDrawer,
+        query,
+        page,
+        PAGE_SIZE,
+        setIsSearching,
+        setHasError,
+        setSearchResult,
+        getLocaleCookie(),
+        selectedTranslations?.length && selectedTranslations.join(','),
+      );
+    },
+    [selectedTranslations],
+  );
+
   useEffect(() => {
     // only when the search query has a value we call the API.
     if (debouncedSearchQuery) {
-      dispatch({ type: addSearchHistoryRecord.type, payload: debouncedSearchQuery });
-      logTextSearchQuery(debouncedSearchQuery, SearchQuerySource.SearchDrawer);
-      setIsSearching(true);
-      getSearchResults({
-        query: debouncedSearchQuery,
-        ...(selectedTranslations &&
-          !!selectedTranslations.length && {
-            filterTranslations: selectedTranslations.join(','),
-          }),
-      })
-        .then((response) => {
-          if (response.status === 500) {
-            setHasError(true);
-          } else {
-            setSearchResult(response);
-            // if there is no navigations nor verses in the response
-            if (response.pagination.totalRecords === 0 && !response.result.navigation.length) {
-              logEmptySearchResults({
-                query: debouncedSearchQuery,
-                source: SearchQuerySource.SearchDrawer,
-              });
-            }
-          }
-        })
-        .catch(() => {
-          setHasError(true);
-        })
-        .finally(() => {
-          setIsSearching(false);
-        });
-    } else {
-      // reset the result
-      setSearchResult(null);
+      addToSearchHistory(dispatch, debouncedSearchQuery, SearchQuerySource.SearchDrawer);
+
+      getResults(debouncedSearchQuery, FIRST_PAGE_NUMBER);
```

Comment:
Why don't we directly do this and remove `getResults` completely since it's used only once with a single changing param?:

```
searchGetResults(
  SearchQuerySource.SearchDrawer,
  debouncedSearchQuery,
  FIRST_PAGE_NUMBER,
  PAGE_SIZE,
  setIsSearching,
  setHasError,
  setSearchResult,
  getLocaleCookie(),
  selectedTranslations?.length && selectedTranslations.join(','),
);